### PR TITLE
fix: Lots of deployment problems [CORE-1985]

### PR DESCRIPTION
**Pull request by roseDickinson**
- Load InstanceID from backends.json
- Points to InstanceID from fixtures by default
- Re-works the store to keep a living vega spec rather than re-generating it every single time
- Replaces urls with proxy urls to avoid CORS issues
- Update styling to be same as latest web app styling
- Remove redundant code
- UX improvements including selecting Geometries from dialog and increasing plot size
- Remove combined file types (csv + geojson / csv + topojson) as they don't currently have the best UX
- Add appropriate loading transitions

### CODE REVIEW

File: api/backends/index.js [hunk 1]
```diff
@@ -4,25 +4,43 @@ import axios from 'axios'
...
     dssauthUrl = response.data.dssauthUrl
     discoveryApiUrl = response.data.discoveryApiUrl
+    nivsMinioUrl = response.data.nivsMinioUrl
+    nidMinioUrl = response.data.nidMinioUrl
     environment = response.data.node_env
+    instanceID = response.data.instanceID
   } catch (error) {
-    console.error('Error while loading settings from server' + error)
+    console.error(`Error while loading settings from server: ${error}`)
+    useDefaultAPIUrls()
```
Comment (tg137) [inline api/backends/index.js:44]: How do the Minio URLs get set if we go down this path? Or is it a case of them not needing to be set?

File: components/Data.vue [hunk 2]
```diff
@@ -325,12 +288,12 @@ export default {
         filename = this.geojsonFiles[this.geojsonIndex].filename
       }
 
-      const res = await axios.get(urlString, {})
+      const res = await downloadFile(urlString)
```
Comment (tg137) [inline components/Data.vue]: Maybe a bit confusing to have a custom `downloadFile` method when there's a `fileDownload` method already? Could change `downloadFile` to `getFileFromMinio` or something?

File: components/Spec.vue [hunk 3]
```diff
@@ -1,4 +1,7 @@
 <template>
```
Comment (tg137) [inline components/Spec.vue]: Is the `template` supposed to be empty here? If so, can we remove the commented code or will we need it?

File: pages/instance/_id.vue [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+</template>
+
+<script>
+import { mapActions, mapMutations, mapState } from 'vuex'
+import Aesthetics from '~/components/Aesthetics'
+import Columns from '~/components/Columns'
+import Geometries from '~/components/Geometries'
+import overlay from '~/components/overlay/Overlay'
+import PlotView from '~/components/PlotView'
+// import Spec from '~/components/Spec'
```
Comment (tg137) [inline pages/instance/_id.vue]: If this is the same empty file from earlier, can't we delete it altogether?

File: api/backends/index.js (hunk 1 above)
Comment (tg137) [reply api/backends/index.js:44]: Just seen that I _think_ they get replaced?

If so, might be worth adding a comment to the nulls above just highlighting that?

**Review by tg137 — CHANGES_REQUESTED** [review]

File: api/backends/index.js (hunk 1 above)
Comment (roseDickinson) [reply api/backends/index.js:44]: useDefaultAPIUrls() sets all the variables to the right thing. The minio urls are variables that actually contain the proxy urls to replace the true minio urls with (in truth its only the hostname not the full url). There is only a cors proxy for staging / production so we only need the urls to be non-null in staging / prod

Comment (roseDickinson) [reply api/backends/index.js:44]: The catch _should_ only happen locally

File: components/Spec.vue (hunk 3 above)
Comment (roseDickinson) [reply components/Spec.vue]: I wasn't sure if we needed the commented code or not - depends on if we are wanting to actually show the text specification to users at some point

File: api/backends/index.js (hunk 1 above)
Comment (tg137) [reply api/backends/index.js:44]: Makes sense, thanks for the comment.

File: components/Spec.vue (hunk 3 above)
Comment (tg137) [reply components/Spec.vue]: Can probably safely drop it I think.

**Review by tg137 — APPROVED** [review]

**Pull request merged**